src/plugins/discourse/mirror.test.js

```diff
@@ -868,4 +877,587 @@ describe("plugins/discourse/mirror", () => {
+      const c1 = fetcher.addCategory();
+      const c2 = fetcher.addCategory();
+      const p3 = fetcher.addPost({topicId: c2.topicId});
+      const t3 = fetcher.addTopic();
+      const t4 = fetcher.addTopic();
+      const p6 = fetcher.addPost({topicId: t4.topicId});
+      await mirror.update(reporter);
+      const posts = [
+        fetcher._post(c1.postId),
+        fetcher._post(c2.postId),
+        fetcher._post(p3),
+        fetcher._post(t3.postId),
+        fetcher._post(t4.postId),
+        fetcher._post(p6),
+      ];
+      expect(repo.posts()).toEqual(posts);
+    });
+
+    it("mirrors updates normal posts, not category definition posts", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+
+      const c1 = fetcher.addCategory();
+      const c2 = fetcher.addCategory();
+      const t3 = fetcher.addTopic();
+      const t4 = fetcher.addTopic();
+      await mirror.update(reporter);
+
+      // Category definition post we don't expect to be updated.
+      fetcher.addPost({topicId: c2.topicId});
+      const p5 = fetcher.addPost({topicId: t4.topicId});
+      await mirror.update(reporter);
+
+      const posts = [
+        fetcher._post(c1.postId),
+        fetcher._post(c2.postId),
+        fetcher._post(t3.postId),
+        fetcher._post(t4.postId),
+        fetcher._post(p5),
+      ];
+      expect(repo.posts()).toEqual(posts);
+    });
+
+    it("mirrors updates normal post edits, not category definition post edits", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+
+      const c1 = fetcher.addCategory();
+      const c2 = fetcher.addCategory();
+      const t3 = fetcher.addTopic();
+      const t4 = fetcher.addTopic();
+      const expectedBefore = [
+        fetcher._post(c1.postId),
+        fetcher._post(c2.postId),
+        fetcher._post(t3.postId),
+        fetcher._post(t4.postId),
+      ];
+      await mirror.update(reporter);
+      const actualBefore = repo.posts();
+
+      // Category definition post we don't expect to be updated.
+      fetcher.editPost(c2.postId, {
+        cooked: `<p>Edit: changed category definition post.</p>`,
+        timestampMs: Date.now(),
+      });
+      fetcher.editPost(t4.postId, {
+        cooked: `<p>Edit: changed normal post.</p>`,
+        timestampMs: Date.now(),
+      });
+      const expectedAfter = [
+        fetcher._post(c1.postId),
+        expectedBefore[1],
+        fetcher._post(t3.postId),
+        fetcher._post(t4.postId),
+      ];
+      await mirror.update(reporter);
+      const actualAfter = repo.posts();
+
+      expect(actualBefore).toEqual(expectedBefore);
+      expect(actualAfter).toEqual(expectedAfter);
+      // Sanity check, make sure the category post was edited in the mock fetcher.
+      expect(fetcher._post(c2.postId)).not.toEqual(expectedAfter[1]);
+    });
+
+    it("mirrors remainder when a post is deleted from the fetcher", async () => {
```

Comment:
"mirrors remainder" -- what does remainder mean in this case?

Reply:
Content exposed by the fetcher which was not deleted.
The remainder after deleting a post from the fetcher.